### PR TITLE
Implement _make_accessor classmethod for PandasDelegate

### PR DESCRIPTION
**Pull request by jbrockmendel**
This is an absolutely minimal subset of the refactor in #17042.  It should be obvious and uncontroversial.

- Define `_make_accessor` as a classmethod of `PandasDelegate`.

 - Make `AccessorProperty.__init__` argument `construct_accessor` default to `accessor_cls._make_accessor`.

- Remove methods `_make_cat_accessor` and `_make_dt_accessor` from `Series`, as they do not belong in the namespace.  Remove `_make_str_accessor` from both `Series` and `Index` namespace.

 - [ ] closes #xxxx
 - [ ] tests added / passed
 - [x] passes ``git diff upstream/master -u -- "*.py" | flake8 --diff``
 - [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/base.py [hunk 1]
```diff
@@ -165,6 +165,12 @@ def __setattr__(self, key, value):
 class PandasDelegate(PandasObject):
     """ an abstract base class for delegating methods/properties """
 
+    @classmethod
+    def _make_accessor(cls, data):
+        raise NotImplementedError("_make_accessor should be implemented"
+                                  "by subclass.  It should return an instance"
+                                  "of `cls`.")
```
Comment (gfyoung) [inline pandas/core/base.py]: Use instead: "_make_accessor should be implemented by subclass and return an instance of `cls`."

File: pandas/core/indexes/accessors.py [hunk 2]
```diff
@@ -243,3 +243,11 @@ class CombinedDatetimelikeProperties(DatetimeProperties, TimedeltaProperties):
...
     # of the more specific classes above.
     __doc__ = DatetimeProperties.__doc__
+
+    @classmethod
+    def _make_accessor(cls, data):
+        try:
+            return maybe_to_datetimelike(data)
+        except Exception:
+            raise AttributeError("Can only use .dt accessor with datetimelike "
+                                 "values")
```
Comment (gfyoung) [inline pandas/core/indexes/accessors.py]: Nit: can we re-organize so that we don't have a lone word on this line?

Comment (jbrockmendel) [inline pandas/core/indexes/accessors.py]: Sure.  I'll take care of this and anything else that piles up this evening.  To the extent that we can exercise self-control, I'm fervently hoping to keep the nitpicking back in #17042.

File: pandas/core/base.py (hunk 1 above)
Comment (jreback) [inline pandas/core/base.py]: we have AbstracMethodError  for this

File: pandas/core/base.py [hunk 3]
```diff
@@ -231,8 +237,11 @@ class AccessorProperty(object):
     """Descriptor for implementing accessor properties like Series.str
     """
 
-    def __init__(self, accessor_cls, construct_accessor):
+    def __init__(self, accessor_cls, construct_accessor=None):
         self.accessor_cls = accessor_cls
+        if construct_accessor is None:
```
Comment (jreback) [inline pandas/core/base.py]: ``self.construct_accessor = construct_accessor or accessor_cls._make_accessor`` is idiomatic

**Review by jreback — APPROVED** [review]
small nit, otherwise lgtm. ping on green.

**Comment by jreback** [comment]
thanks!

**Pull request merged**